### ISSUE TITLE
[bug] Reconnecting to Coinbase Wallet doesn't work without refreshing the page

### ISSUE THREAD

**Issue by iketw** [rainbow-me/rainbowkit#1087]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

0.12.1

### wagmi Version

0.12.1

### Current Behavior

1. Connect to Coinbase Wallet by scanning QR code with mobile app
2. Disconnect on the website without refreshing the page
3. Try to connect again: The mobile app just scans and closes the camera, no connection is established and the qr code stays up in the website

### Expected Behavior

I'd expect to connect/reconnect as many times as needed without having to refresh the page

### Steps To Reproduce

1. Connect to Coinbase Wallet by scanning QR code with mobile app
2. Disconnect on the website without refreshing the page
3. Try to connect again: The mobile app just scans and closes the camera, no connection is established and the qr code stays up in the website

**Comment by DanielSinclair** [comment]
@iketw I'm actually not able to replicate this. I tested with the latest Coinbase Wallet app on iOS and our test dApp on Chrome. I was able to successfully connect and disconnect multiple times in succession. This sounds like a problem with the Coinbase Wallet app specifically, as wallets implement the other half of the pairing handshake. If a wallet fails to pair, RainbowKit would be stuck in the state you describe waiting for the pairing to complete. What platform are you on? Did you do anything differently, like switching wallet accounts before attempting to reconnect?

**Comment by iketw** [comment]
> @iketw I'm actually not able to replicate this. I tested with the latest Coinbase Wallet app on iOS and our test dApp on Chrome. I was able to successfully connect and disconnect multiple times in succession. This sounds like a problem with the Coinbase Wallet app specifically, as wallets implement the other half of the pairing handshake. If a wallet fails to pair, RainbowKit would be stuck in the state you describe waiting for the pairing to complete. What platform are you on? Did you do anything differently, like switching wallet accounts before attempting to reconnect?

@DanielSinclair, thanks for taking a look into this. I tested with the Coinbase Wallet on an Android device (version: 28.32.0)

"@rainbow-me/rainbowkit": "^0.12.1"
"ethers": "^5.1.0",
"wagmi": "^0.12.1"

I didn't do anything other than connect, disconnect, try to connect again

https://user-images.githubusercontent.com/121973632/226377470-9e9c274a-8526-4b43-82bf-d22072d086f8.mov

I can report the bug to Coinbase instead then if you think this may be a bug with their wallet.

**Comment by DanielSinclair** [comment]
Ah, this does look like an issue. There must be something wrong with the Coinbase's underlying connector; you shouldn't be seeing that modal appear on top of RainbowKit. The provider and QR code is not being refreshed properly, and thus the behavior you're seeing on the mobile wallet side. Will investigate~

**Comment by saajand** [comment]
I have encountered the same issue as well. And get some additional issues along with it.
Sometimes even on the first connect Rainbowkit modal does not responds, stays showing the QR, then I have to refresh the page and the wallet would already be connected then.
But Sometimes it won't be connected even after a refresh. Closing and reopening the Coinbase wallet App helps in this case.

"@rainbow-me/rainbowkit": "^0.12.1"
"ethers": "5.7.2"
"wagmi": "^0.12.1"
Coinbase Wallet App: Version 28.32.0 (Android)
